sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/SubquerySuite.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.catalyst.dsl.expressions._
+import org.apache.spark.sql.catalyst.plans.logical.LocalRelation
+
+class SubquerySuite extends SparkFunSuite  {
+
+  test("SPARK-27246 test the scalar subquery with no columns ") {
+    val e = intercept[AssertionError] {
+      ScalarSubquery(LocalRelation()).as('a).treeString
```

Comment:
wouldn't calling dataType be a more direct way of triggering this failure ?

Reply:
yes it can be, i just tried to use the same way as per the testcase whcih I removed for this PR

Reply:
@sandeep-katta oh...but the removed testcase does not have a call to `treeString`, right ? the treeString happens when we turn on debugging of the plans , no ?

Reply:
yes

Reply:
Let's write that in the PR description

Reply:
updated